### PR TITLE
SDI-184 enabling tests and implementation for return from court movem…

### PR DESCRIPTION
**Pull request by lbennett-moj**
…ents (different prison)

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/transfer/PrisonTransferService.kt [hunk 1]
```diff
@@ -125,12 +155,23 @@ class PrisonTransferService(
...
-    val internalLocationCode = cellLocation ?: "${prison.id}-RECP"
+    return getCellLocation(cellLocation, prison.id)
+  }
+
+  private fun getCellLocation(cellLocation: String?, prisonCode: String): Result<AgencyInternalLocation> {
+    val internalLocationCode = cellLocation ?: "$prisonCode-RECP"
     return agencyInternalLocationRepository.findOneByDescriptionAndAgencyId(
-      internalLocationCode, prison.id
+      internalLocationCode, prisonCode
     ).map { success(it) }
```
Comment (lbennett-moj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/transfer/PrisonTransferService.kt:165]: will change this to prisonId to be fairly consistent

File: src/main/java/uk/gov/justice/hmpps/prison/service/transfer/PrisonTransferService.kt [hunk 2]
```diff
@@ -125,12 +155,23 @@ class PrisonTransferService(
   }
 
   private fun getCellLocation(cellLocation: String?, prison: AgencyLocation): Result<AgencyInternalLocation> {
-    val internalLocationCode = cellLocation ?: "${prison.id}-RECP"
+    return getCellLocation(cellLocation, prison.id)
+  }
+
+  private fun getCellLocation(cellLocation: String?, prisonCode: String): Result<AgencyInternalLocation> {
+    val internalLocationCode = cellLocation ?: "$prisonCode-RECP"
     return agencyInternalLocationRepository.findOneByDescriptionAndAgencyId(
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/transfer/PrisonTransferService.kt:163]: in the future we can look at whether there is a better way of finding a prison's reception location

**Review by andymarke — DISMISSED** [review]
LGTM

**Review by andymarke — APPROVED** [review]
LGTM

**Review by andymarke — APPROVED** [review]

**Pull request merged**